io/directory.go

```diff
@@ -139,6 +148,33 @@ func NewDirectoryFromNode(dserv ipld.DAGService, node ipld.Node) (Directory, err
 	return nil, ErrNotADir
 }
 
+func (d *BasicDirectory) computeEstimatedSize() {
+	d.ForEachLink(nil, func(l *ipld.Link) error {
+		d.addToEstimatedSize(l.Name, l.Cid)
+		return nil
+	})
+}
+
+func estimatedLinkSize(linkName string, linkCid cid.Cid) int {
+	return len(linkName) + linkCid.ByteLen()
+}
+
+func (d *BasicDirectory) addToEstimatedSize(name string, linkCid cid.Cid) {
+	d.estimatedSize += estimatedLinkSize(name, linkCid)
+	// FIXME: Ideally we may want to track the Link size as well but it is
```

Comment:
Let's remove this FIXME, we don't want someone to actually come along and fix this in code (it's a spec change, effectively.